dasharo-stability/capsule-update.robot

```diff
@@ -0,0 +1,96 @@
+*** Settings ***
+Library             Collections
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+# TODO: maybe have a single file to include if we need to include the same
+# stuff in all test cases
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+
+# TODO:
+# - document which setup/teardown keywords to use and what are they doing
+# - go threough them and make sure they are doing what the name suggest (not
+# exactly the case right now)
+Suite Setup         Run Keywords
+...                     Prepare Test Suite
+...                     AND
+...                     Skip If    not ${CAPSULE_UPDATE_SUPPORT}    Capsule Update not supported
+...                     AND
+...                     Flash Firmware If Not QEMU
+Suite Teardown      Run Keywords
+...                     Flash Firmware If Not QEMU
+...                     AND
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+CUP001.001 Capsule Update (Ubuntu)
+    [Documentation]    This test aims to verify if Capsule Update process works.
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+
+    Set Prompt For Terminal    root@${UBUNTU_HOSTNAME}:/home#
+    Execute Command In Terminal    cd ..
+    Set Prompt For Terminal    root@${UBUNTU_HOSTNAME}:/#
+    Execute Command In Terminal    cd ..
+    Execute Command In Terminal    rm -r capsule_testing
+    Execute Command In Terminal    mkdir capsule_testing
+    Execute Command In Terminal    chmod 777 capsule_testing
+
+    Send File To DUT    ./dasharo-stability/capsule-update-files/CapsuleApp.efi    /capsule_testing/CapsuleApp.efi
+    Send File To DUT    ${CAPSULE_FW_FILE}    /capsule_testing/dasharo.cap
+
+    Power On
+    ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
+    Enter Submenu From Snapshot    ${boot_menu}    UEFI Shell
+    Read From Terminal Until    Shell>
+
+    ${out}=    Execute UEFI Shell Command    smbiosview -t 0
+    ${original_bios_version}=    Extract BIOS Version    ${out}
+    Execute UEFI Shell Command    FS0:
+    Execute UEFI Shell Command    cd capsule_testing
+    Execute UEFI Shell Command    CapsuleApp.efi dasharo.cap
+
+    ${out}=    Read From Terminal Until    seconds.)
```

Comment:
Full output is needed as there is a need to press a particular key (which changes every time) to continue the operation.

Reply:
What I meant is something like:

```suggestion
    Read From Terminal Until    This message is displayed because the platform has booted in Firmware
    ${out}=    Read From Terminal Until    seconds.)
```

On debug binaries, there will be a lot of text before this popup, and it will definitely have numbers in it. You don't need whole output, just the part with key number.

Reply:
@SergiiDmytruk please close the threads if you think they have been addressed.

Reply:
> JanPrusinowski marked this conversation as resolved.

Please do not resolve threads when you are the assignee. Let the reviewers decide whether the review has been properly addressed.